aten/src/ATen/native/cuda/ForeachBinaryOpScalarTensor.cu

```diff
@@ -0,0 +1,147 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Dispatch.h>
+#include <ATen/native/BinaryOps.h>
+#include <ATen/native/ForeachUtils.h>
+#include <ATen/native/cuda/ForeachFunctors.cuh>
+#include <ATen/native/cuda/ForeachMinMaxFunctors.cuh>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/_foreach_mul_native.h>
+
+#include <ATen/ops/empty_like_native.h>
+#endif
+
+namespace at::native {
+
+template <typename T, template <class> class Op>
+std::vector<Tensor> foreach_binary_op(TensorList tensors, const Tensor& other) {
```

Comment:
```suggestion
std::vector<Tensor> foreach_binary_op(TensorList tensors, const Tensor& scalar) {
```
This is a nit that you can feel free to ignore, but do you think it may be clearer to name the Tensor scalar to show the expectation of it being a scalar to a code reader?